components/brave_extension/extension/brave_extension/content_cosmetic.ts

```diff
@@ -249,41 +261,73 @@ const isSubTreeFirstParty = (elm: Element, possibleQueryResult?: IsFirstPartyQue
     return false
   }
   const htmlElement = asHTMLElement(elm)
-  if (!htmlElement || !htmlElement.innerText.trim().length) {
+  if (!htmlElement || isAdText(htmlElement.innerText) === false) {
     return false
   }
   return true
 }
 
-const hideSelectors = (selectors: string[]) => {
-  if (selectors.length === 0) {
+const unhideSelectors = (selectors: Set<string>) => {
+  if (selectors.size === 0) {
     return
   }
-
-  chrome.runtime.sendMessage({
-    type: 'hideThirdPartySelectors',
-    selectors
-  })
+  const origRuleCount = cosmeticStyleSheet.rules.length
+  // Find selectors we have a rule index for
+  const rulesToRemove = Array.from(selectors)
+    .map(selector =>
+      allSelectorsToRules.get(selector)
+    )
+    .filter(i => i !== undefined)
+    .sort()
+    .reverse()
+  // Delete the rules
+  let lastIdx: number = allSelectorsToRules.size - 1
+  for (const ruleIdx of rulesToRemove) {
+    cosmeticStyleSheet.deleteRule(ruleIdx)
+  }
+  // Re-sync the indexes
+  // TODO: Sync is hard, just re-build by iterating through the StyleSheet rules.
+  const ruleLookup = Array.from(allSelectorsToRules.entries())
+  let countAtLastHighest = rulesToRemove.length
+  for (let i = lastIdx; i > 0; i--) {
+    const [selector, oldIdx] = ruleLookup[i]
+    // Is this one we removed?
+    if (rulesToRemove.includes(i)) {
+      allSelectorsToRules.delete(selector)
+      countAtLastHighest--
+      if (countAtLastHighest === 0) {
+        break
+      }
+      continue
+    }
+    if (oldIdx !== i) {
```

Comment:
similarly, whats going on here?

Reply:
A sanity check that any previous re-indexes have kept this item in-sync. Though since we expect the actual `index` of an item in the `Map` to match the lookup value of the Map's item, we probably may as well use a combination of `Set` and `array`. Either way we would be doing something like converting Set to Array to find an index, or using `indexOf` for the lookup. So, this works for now and seems just as performant. Having said that, would love to optimize if you can think of a way.